it-ops-shared/ITOps.ViewModelComposition.Gateway/ApplicationBuilderExtensions.cs

```diff
@@ -13,14 +15,20 @@ public static void RunCompositionGateway(this IApplicationBuilder app, Action<IR
             var routeBuilder = new RouteBuilder(app);
             routes?.Invoke(routeBuilder);
 
+            var allModules = app.ApplicationServices.GetServices<RequestsModule>();
```

Comment:
Wouldn't this line instantiate all handlers with all their dependencies? In a relatively big api, this can be a lot of stuff. Attribute routing would solve it. Also the approach I mentioned before Get<THandler> as you wouldn't instantiate the handlers.

Reply:
the problem is that the routing engine needs to have all the routing information beforehand, so if you want to use Nancy style modules they need to be instantiated, paying this price. If you want to use attribute routing there will be the reflection price. The current approach, despite being the most verbose (Matches+Handle) is the most flexible and cheapest one,m at the cost of verbosity.

Reply:
Yes, that's why I suggested using `Get<THandle>(route)` instead of `Get(route, lambda)`, because then you can instantiate all modules to get the routes without instantiating the handlers. The implementation should be very simple: instead of calling the lambda, resolve THandle with the container and call Handle();